module/zfs/spa_errlog.c

```diff
@@ -136,9 +533,60 @@ spa_log_error(spa_t *spa, const zbookmark_phys_t *zb)
 uint64_t
 spa_get_errlog_size(spa_t *spa)
 {
-	uint64_t total = 0, count;
+	uint64_t total = 0;
+
+	if (!spa_feature_is_enabled(spa, SPA_FEATURE_HEAD_ERRLOG)) {
+		uint64_t count;
+		mutex_enter(&spa->spa_errlog_lock);
+		if (spa->spa_errlog_scrub != 0 &&
+		    zap_count(spa->spa_meta_objset, spa->spa_errlog_scrub,
+		    &count) == 0)
+			total += count;
+
+		if (spa->spa_errlog_last != 0 && !spa->spa_scrub_finished &&
+		    zap_count(spa->spa_meta_objset, spa->spa_errlog_last,
+		    &count) == 0)
+			total += count;
+		mutex_exit(&spa->spa_errlog_lock);
+
+		mutex_enter(&spa->spa_errlist_lock);
+		total += avl_numnodes(&spa->spa_errlist_last);
+		total += avl_numnodes(&spa->spa_errlist_scrub);
+		mutex_exit(&spa->spa_errlist_lock);
+
+	} else {
+#ifdef _KERNEL
+		mutex_enter(&spa->spa_errlog_lock);
+		total += get_errlog_size(spa, spa->spa_errlog_last);
+		total += get_errlog_size(spa, spa->spa_errlog_scrub);
+		mutex_exit(&spa->spa_errlog_lock);
+
+		mutex_enter(&spa->spa_errlist_lock);
+		total += get_errlist_size(spa, &spa->spa_errlist_last);
+		total += get_errlist_size(spa, &spa->spa_errlist_scrub);
+		mutex_exit(&spa->spa_errlist_lock);
+#endif
+	}
+	return (total);
+}
+
+#ifdef _KERNEL
```

Comment:
We should enable all of this code in both the user and kernel builds do it's available as needed for tools like `ztest` and `zhack`.  Unless that's not possible for some reason?